### PR TITLE
Adds multiply and divide aliases

### PR DESCRIPTION
**Pull request by mruberry**
These alias are consistent with NumPy. Note that C++'s naming would be different (std::multiplies and std::divides), and that PyTorch's existing names (mul and div) are consistent with Python's dunders. 

This also improves the instructions for adding an alias to clarify that dispatch keys should be removed when copying native_function.yaml entries to create the alias entries.

### CODE REVIEW

File: test/test_op_aliases.py [hunk 1]
```diff
@@ -128,6 +128,22 @@ def __init__(self,
               lambda d: torch.randn(20, device=d),
               get_args=lambda d: (torch.randn(20, device=d),),
               decorators=(onlyCPU,)),
+    # NOTE: only runs on CPU because it leaks CUDA memory
+    #   (see https://github.com/pytorch/pytorch/issues/43119)
+    AliasInfo('divide', torch.divide, 'div', torch.div,
+              lambda d: torch.randn(20, device=d), get_args=lambda d: (torch.rand(20, device=d) + .1,),
+              decorators=(onlyCPU,)),
```
Comment (gchanan) [inline test/test_op_aliases.py:135]: so we check CUDA memory leaks but not CPU?  Or it doesn't leak on CPU?

File: aten/src/ATen/native/BinaryOps.cpp [hunk 2]
```diff
@@ -259,6 +294,35 @@ Tensor& mul_(Tensor& self, const Tensor& other) {
   return native::mul_out(self, self, other);
 }
 
+Tensor mul(const Tensor& self, Scalar other) {
+  return native::mul(self, wrapped_scalar_tensor(other));
```
Comment (gchanan) [inline aten/src/ATen/native/BinaryOps.cpp:298]: is this correct?  Won't this skip dispatch which means you can't dispatch to the sparse variant?

python might avoid this by always calling the tensor overload, but maybe it's an issue in C++?

File: aten/src/ATen/native/BinaryOps.cpp [hunk 3]
```diff
@@ -259,6 +294,35 @@ Tensor& mul_(Tensor& self, const Tensor& other) {
...
+}
+
+Tensor& mul_(Tensor& self, Scalar other) {
+  return native::mul_(self, wrapped_scalar_tensor(other));
```
Comment (gchanan) [inline aten/src/ATen/native/BinaryOps.cpp:302]: here too.

**Review by gchanan — APPROVED** [review]

File: aten/src/ATen/native/BinaryOps.cpp (hunk 2 above)
Comment (mruberry) [reply aten/src/ATen/native/BinaryOps.cpp:298]: You're probably right. This is existing code that I just moved next to the other mul functions for consistency. I've made a note to review this behavior.

File: test/test_op_aliases.py (hunk 1 above)
Comment (mruberry) [reply test/test_op_aliases.py:135]: We don't check CPU memory leaks. This will be fixed when the tests are refactored.

**Pull request closed without merging**